src/main/java/me/zeroeightsix/kami/module/modules/combat/CrystalAura.java

```diff
@@ -61,9 +68,32 @@ public void onUpdate() {
                 .map(entity -> (EntityEnderCrystal) entity)
                 .min(Comparator.comparing(c -> mc.player.getDistance(c)))
                 .orElse(null);
-        if (crystal != null && mc.player.getDistance(crystal) <= range.getValue()) {
+        if (explode.getValue() && crystal != null && mc.player.getDistance(crystal) <= range.getValue()) {
             //Added delay to stop ncp from flagging "hitting too fast"
             if (((System.nanoTime() / 1000000) - systemTime) >= 250) {
+                if (antiweakness.getValue() && mc.player.isPotionActive(MobEffects.WEAKNESS)) {
+                    // search blocks in hotbar
+                    newSlot = -1;
+                    for (int i = 0; i < 9; i++) {
+                        ItemStack stack = Wrapper.getPlayer().inventory.getStackInSlot(i);
+                        if (stack == ItemStack.EMPTY) {
+                            continue;
+                        }
+                        if ((stack.getItem() instanceof ItemSword)) {
+                            newSlot = i;
+                            break;
+                        }
+                        if ((stack.getItem() instanceof ItemTool)) {
+                            newSlot = i;
+                            break;
+                        }
+                    }
+                    // check if any swords or tools were found
+                    if (newSlot != -1) {
+                        Wrapper.getPlayer().inventory.currentItem = newSlot;
+                        switchCooldown = true;
+                    }
+                }
```

Comment:
What's this feature for?
Weakness reduces attack damage - crystals the player hit will remain unaffected in damage. Switching to a sword (or another tool) doesn't make any sense, does it?

Reply:
Indeed with a Sword or Tool a Player can explode crystals while having the Weakness Effect.

Reply:
What's keeping the player from just attacking the crystal with the item they're already holding now? Why do we switch to something else?

Reply:
While under the influence of Weakness, only with a Sword or Tool (Items with additional Attack Damage) the Player is able to break crystals, with a bare Hand or a Stack of Crystals or Obi in Hand, the attacked crystal will not break.